collect_repos.py

```diff
@@ -0,0 +1,75 @@
+'''
+This script collects the current state of all student repositories for a given assignment in a given organization 
+Run this script from inside a directory that contains/will contain all the repos for this assignment
+'''
+
+from github3 import login
+import getpass
+import subprocess
+import csv
+import configparser
+import os
+
+def run(username, token, organization, prefix, list, deadline):
+	with open(token, 'r') as f:
+		token = f.readline().strip()
+
+	github = login(username,token=token)
+	memberships = github.organization_memberships()
+	course = None
+
+	for membership in memberships:
+		if membership.organization.login.lower() == organization.lower():
+			course = membership.organization
+			break
+
+	newDir = os.getcwd() + "/{}".format(prefix) #pull all repos to this folder
+	if not os.path.exists(newDir):
+		os.mkdir(newDir)
+
+	with open('student_list.csv', mode='r') as f:
+		reader = csv.reader(f)
+		mapping = {rows[1]:rows[2] for rows in reader}
+
+	if list:
+		with open(list, 'r') as f:
+			ccids = [ccid.strip() for ccid in f.readlines()]
+			usernames = [mapping[ccid] for ccid in ccids]  #convert ccid to github username
+	else:
+		usernames = []
+
+	repos = []
+	for repo in course.repositories():
+		if len(usernames):
+			if repo.name.replace('{}-'.format(prefix), '') in usernames:
+				repos.append(repo)
+		elif repo.name.startswith(prefix):
+			repos.append(repo)
```

Comment:
I guess we can ignore for now (and that's probably what i had from before), but in terms of efficiency, is there a reason we need to loop twice on the repos? Once we know we should consider a repo, it might make sense to go ahead and process it?

Reply:
I just want to avoid duplicate code, I can make it like this, if you prefer.

<img width="1033" alt="Screen Shot 2021-01-14 at 11 02 33 AM" src="https://user-images.githubusercontent.com/16128272/104630505-0d965980-5658-11eb-8d43-a270a8697c0e.png">

Reply:
why would you have to duplicate the code? Once you have repoDir, wouldn't the rest be the same code and can come after the main if/elif?

Reply:
I put them into a new function, it should be fine now.